### ISSUE TITLE
💡 [REQUEST] - What is purpose of `out.backward(torch.randn(1, 10))` in neural_networks_tutorial

### ISSUE THREAD

**Issue by Lovkush-A** [pytorch/tutorials#3017]
### 🚀 Describe the improvement or the new tutorial

In [neural networks tutorial for beginners](https://pytorch.org/tutorials/beginner/blitz/neural_networks_tutorial.html), we have the following:

Zero the gradient buffers of all parameters and backprops with random gradients:
```
net.zero_grad()
out.backward(torch.randn(1, 10))
```

What is the purpose of this? It is not part of standard ML workflows and can be confusing to beginners. (As evidence,I am helping some people learn basics of ML and I got questions about this line. This is how I found out about it!)

If there is no good reason for it, then I suggest:
- dropping these few lines
- changing wording of other parts of the page if needed. E.g. 'at this point we covered... calling backward'

**Comment by albanD** [comment]
I would agree the random gradient can be confusing if you're not already familiar with how backprop work.
`out.sum().backward()` might be less confusing here?

**Comment by Lovkush-A** [comment]
@albanD What is the downside of just dropping doing backward in this cell? I think `out.sum().backward()` is also confusing because it is not part of the standard ml workflow.

**Comment by albanD** [comment]
Not sure, given some of the other wording in that tutorial, it might come from an earlier iteration where backward was discussed at the beginning.

**Comment by Lovkush-A** [comment]
Makes sense that it is from a previous version of the tutorial.

Can I make a PR to drop this reference to backward then?

**Comment by albanD** [comment]
I'll let @svekars help on what we want to do to update the general outline here.

**Comment by svekars** [comment]
Please feel free to submit a PR with these changes.
